sdn_tests/testplans/SUBSCRIBE-Modes-Levels-Test-Plan.md

```diff
@@ -0,0 +1,2150 @@
+-   STREAM
+    -   ON_CHANGE
+        -   Only certain nodes support this mode. Provides an updated response that is triggered by an update to the backend database.
+    -   SAMPLE
+        -   Provides an updated response that is triggered by a timer.
+    -   TARGET_DEFINED
+        -   A mixture of ON_CHANGE and SAMPLE updates. Preference is for ON_CHANGE if it is supported.
+
+-   Optional Flags
+    -   Updates_only - Does not provide initial data
+    -   Heartbeat_interval - Provide full data based on timer. Used to override other flags.
+    -   Suppress_redundant - Do not provide data that has not changed since last update.
+
+Not all modes are supported for a subscription level/point. Below is a summary of the expected support at each level as well as different optional flags:
+
+<table>
+  <thead>
+    <tr>
+      <th></th>
+      <th><br>
+ONCE</th>
+      <th><br>
+POLL</th>
+      <th><br>
+ON_CHANGE</th>
+      <th><br>
+SAMPLE</th>
+      <th><br>
+TARGET_DEFINED</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td><br>
+Root </td>
+      <td><br>
+Yes</td>
+      <td><br>
+Yes</td>
+      <td><br>
+No</td>
+      <td><br>
+Yes</td>
+      <td><br>
+Yes (same as SAMPLE)</td>
+    </tr>
+    <tr>
+      <td><br>
+Subtree</td>
+      <td><br>
+YES</td>
+      <td><br>
+YES</td>
+      <td><br>
+No</td>
+      <td><br>
+Yes</td>
+      <td><br>
+Yes (same as SAMPLE)</td>
+    </tr>
+    <tr>
+      <td><br>
+Leaf</td>
+      <td><br>
+Yes</td>
+      <td><br>
+Yes</td>
+      <td><br>
+Depends on Requirements</td>
+      <td><br>
+Yes</td>
+      <td><br>
+Yes (can be SAMPLE or ON_CHANGE depends on the node)</td>
+    </tr>
+    <tr>
+      <td><br>
+updates_only</td>
+      <td><br>
+yes</td>
+      <td><br>
+yes</td>
+      <td><br>
+yes</td>
+      <td><br>
+yes</td>
+      <td><br>
+yes</td>
+    </tr>
+    <tr>
+      <td><br>
+heartbeat_interval</td>
+      <td><br>
+no</td>
+      <td><br>
+no</td>
+      <td><br>
+yes</td>
+      <td><br>
+yes</td>
+      <td><br>
+yes</td>
+    </tr>
+    <tr>
+      <td><br>
+suppress_redundant</td>
+      <td><br>
+no</td>
+      <td><br>
+no</td>
+      <td><br>
+no</td>
+      <td><br>
+Yes</td>
+      <td><br>
+Yes (if it is a Sample leaf)</td>
+    </tr>
+  </tbody>
+</table>
+
+For TARGET_DEFINED subscription, current UMF implementation will use SAMPLE on the whole subtree if any node in the subtree doesn't support ON_CHANGE. So for root and subtree level TARGET_DEFINED subscription, only SAMPLE is covered. Mixed-use of ON_CHANGE and SAMPLE may be added in the future if it is required and implemented.
+
+# gNMI Feature Requirements to be tested
+
+To achieve optionality, it is important to address any gaps to the gNMI specification in the GPINS program. This test plan focuses on the verification of functional aspects of the Subscribe RPC of different types only. Other gNMI features are covered in different GPINS test plans.\
```

Comment:
It is important to address any gaps to the gNMI specification in Sonic-Gnmi. This test plan focuses on the verification of functional aspects of the Subscribe RPC of different types and levels. Other gNMI features are covered in other test plans.